### PR TITLE
openjdk: add patches for musl

### PR DESCRIPTION
**Pull request by xmhd**
@gyakovlev 

Add patches to OpenJDK 8 and OpenJDK 11 ebuilds for musl libc compatibility.

Patches are from Alpine Linux and tested on ~amd64.

### CODE REVIEW

File: dev-java/openjdk/openjdk-11.0.8_p10.ebuild [hunk 1]
```diff
@@ -72,6 +72,13 @@ REQUIRED_USE="javafx? ( alsa !headless-awt )"
 
 S="${WORKDIR}/jdk${SLOT}u-jdk-${MY_PV}"
 
+PATCHES=(
+        "${FILESDIR}/patches/${PN}-${SLOT}/build.patch"
+        "${FILESDIR}/patches/${PN}-${SLOT}/fix-bootjdk-check.patch"
+        "${FILESDIR}/patches/${PN}-${SLOT}/ppc64le.patch"
+        "${FILESDIR}/patches/${PN}-${SLOT}/aarch64.patch"
```
Comment (gyakovlev) [inline dev-java/openjdk/openjdk-11.0.8_p10.ebuild:79]: please use tabs

File: dev-java/openjdk/openjdk-11.0.8_p10.ebuild [hunk 2]
```diff
@@ -131,6 +138,14 @@ pkg_setup() {
 
 src_prepare() {
 	default
+
+
+	# remove not compilable module (hotspot jdk.hotspot.agent)
+	# this needs libthread_db which is only provided by glibc
+	#
+	# haven't found any way to disable this module so just remove it.
+        rm -rf "${S}"/src/jdk.hotspot.agent
```
Comment (gyakovlev) [inline dev-java/openjdk/openjdk-11.0.8_p10.ebuild:147]: maybe guard it by `use elibc_musl && { rm "${S}"/src/jdk.hotspot.agent || die ; }`
note the missing `die`
also use tabs.

File: dev-java/openjdk/openjdk-8.265_p01.ebuild [hunk 3]
```diff
@@ -91,6 +91,19 @@ DEPEND="
...
+        "${FILESDIR}/patches/${PN}-${SLOT}/0001_musl_hotspot.patch"
+        "${FILESDIR}/patches/${PN}-${SLOT}/0002_musl_hotspot_ppc.patch"
+        "${FILESDIR}/patches/${PN}-${SLOT}/0003_musl_hotspot_aarch64.patch"
+        "${FILESDIR}/patches/${PN}-${SLOT}/0004_musl_hotspot_noagent.patch"
+        "${FILESDIR}/patches/${PN}-${SLOT}/0005_musl_fix_libjvm_load.patch"
+        "${FILESDIR}/patches/${PN}-${SLOT}/0006_musl_jdk.patch"
+        "${FILESDIR}/patches/${PN}-${SLOT}/0007_musl_jdk_includes.patch"
+        "${FILESDIR}/patches/${PN}-${SLOT}/0008_musl_jdk_execinfo.patch"
+        "${FILESDIR}/patches/${PN}-${SLOT}/0009_fix_jdk_ipv6_init.patch"
+        "${FILESDIR}/patches/${PN}-${SLOT}/0010_fix_jdk_close_fds.patch"
```
Comment (gyakovlev) [inline dev-java/openjdk/openjdk-8.265_p01.ebuild:104]: tabs

File: dev-java/openjdk/openjdk-8.265_p01.ebuild [hunk 4]
```diff
@@ -139,6 +152,11 @@ src_prepare() {
 	sed -i '/^WARNINGS_ARE_ERRORS/ s/-Werror/-Wno-error/' \
 		hotspot/make/linux/makefiles/gcc.make || die
 
+        # apply this patch here as the sources are not available unless ARCH == arm64
+        if use arm64; then
+            eapply "${FILESDIR}/patches/${PN}-${SLOT}/0003_musl_hotspot_aarch64.patch"
+        fi
```
Comment (gyakovlev) [inline dev-java/openjdk/openjdk-8.265_p01.ebuild:158]: also tabs

**Review by gyakovlev — CHANGES_REQUESTED** [review]
I haven't checked but I have a feeling patches are quite large.
did you run repoman? it would have warned if so.

also missing `Signed-off-by`

**Comment by rossbridger** [comment]
Think it might be better suited for the musl overlay?

**Comment by gyakovlev** [comment]
> Think it might be better suited for the musl overlay?

quite maybe. I need to test it first myself.
as jdk maintainer I don't mind those patches in teh main tree, but in a way it does not break regular users and does not block security bumps.
we may move it to musl initially for baking-in.

**Pull request closed without merging**